modules/promotion/src/Element/CouponRedemptionForm.php

```diff
@@ -144,7 +171,20 @@ public static function processForm(array $element, FormStateInterface $form_stat
   public static function ajaxRefresh(array $form, FormStateInterface $form_state) {
     $parents = $form_state->getTriggeringElement()['#parents'];
     array_pop($parents);
-    return NestedArray::getValue($form, $parents);
+    $coupon_element = NestedArray::getValue($form, $parents);
+
+    $response = new AjaxResponse();
+    // To refresh the coupon.
+    $response->addCommand(new InsertCommand(NULL, $coupon_element));
+    // To refresh the order summary.
+    // @todo the checkout pane should handle this, not the element.
+    if (isset($form['sidebar']['order_summary'])) {
```

Comment:
I want to find an elegant way to attach extra callbacks here. Maybe `#element_ajax`?